upgrade/scripts/upgrade/prerequisites.sh

```diff
@@ -833,11 +833,34 @@ has_craycm() {
   helm list -n "${ns}" --filter 'cray-certmanager$' | grep cray-certmanager > /dev/null 2>&1
 }
 
-state_name="UPGRADE_CERTMANAGER_CHART"
+state_name="UPGRADE_CERTMANAGER_0141_CHART"
 state_recorded=$(is_state_recorded "${state_name}" "$(hostname)")
 if [[ ${state_recorded} == "0" && $(hostname) == "${PRIMARY_NODE}" ]]; then
   echo "====> ${state_name} ..." | tee -a "${LOG_FILE}"
   {
+    # The actions below only need to happen iff we are on 0.14.1 if we're
+    # already at 1.5.5+ the crd ownership changes makes this logic impossible to
+    # work due to helm hooks. Making this work on both isn't really worth the
+    # time so just constrain this block of logic to 0.14.1 where we know its
+    # needed.
+    gate="0.14.1"
+    found=$(helm list -n cert-manager --filter 'cray-certmanager$' | awk '/deployed/ {print $10}')
+
+    if [ "${found}" = "${gate}" ]; then
+      printf "note: found old cert-manager that needs upgrades %s\n" "${found}" >&2
+    else
+      printf "note: cert-manager helm chart version %s\n" "${found}" >&2
+
+      # We might be rerunning from a pre 1.5.x install and there is no
+      # cert-manager installed due to a prior removal
+      if [ "${found}" = "" ]; then
+	printf "note: no helm install appears to exist for cert-manager, likely this state is being run again\n" >&2
```

Comment:
needs to fix the indent

Reply:
Explain, its 6 spaces and 3 indent levels deep:

```
$ grep 'note: found' upgrade/scripts/upgrade/prerequisites.sh | xxd
00000000: 2020 2020 2020 7072 696e 7466 2022 6e6f        printf "no
00000010: 7465 3a20 666f 756e 6420 6f6c 6420 6365  te: found old ce
00000020: 7274 2d6d 616e 6167 6572 2074 6861 7420  rt-manager that 
00000030: 6e65 6564 7320 7570 6772 6164 6573 2025  needs upgrades %
00000040: 735c 6e22 2022 247b 666f 756e 647d 2220  s\n" "${found}" 
00000050: 3e26 320a                                >&2.
```

Reply:
ah wrong one, fixed.